Sources/GameCore/GameCore.swift

```diff
@@ -1084,8 +1084,19 @@ extension Reducer where State == GameState, Action == GameAction, Environment ==
         .fireAndForget()
 
     case let .gameCenter(.turnBasedMatchResponse(.success(match))):
-      // TODO: Should this replace `GameState`?
-      state.turnBasedContext?.match = match
```

Comment:
This was the culprit. If you play your first turn in a game and immediately quit/forfeit, the submitted game had stale match metadata.